controller/array_action/array_mediator_ds8k.py

```diff
@@ -517,15 +517,16 @@ def get_snapshot(self, volume_id, snapshot_name, pool=None):
             return None
         return self._generate_snapshot_response_with_verification(api_snapshot)
 
-    def _create_similar_volume(self, target_volume_name, source_api_volume, pool_id):
+    def _create_similar_volume(self, target_volume_name, source_api_volume, space_efficiency, pool):
         logger.info(
             "creating target api volume '{0}' from source volume '{1}'".format(target_volume_name,
                                                                                source_api_volume.name))
-        space_efficiency = source_api_volume.tp
+        if not space_efficiency:
```

Comment:
IMO `if space_efficiency is None:` here would be better, to distinguish between specified-empty SE and unspecified SE

Reply:
I don't see what `specified-empty pool` means.
agreed about the SE

Reply:
for explicit empty pool, we should consider let the user get an error from the storage (and not take the source pool in such case)

Reply:
nevermind, let's treat both as "unspecified" and remove `None`s from the interface, the server doesn't need the "optionality" anyway

Reply:
for consistency:
empty parameters will be equal in behavior to non-specified.

Reply:
right

Reply:
done

Reply:
so could you remove the default values in `create_snapshot`?

Reply:
done

Reply:
including the [implementation](https://github.com/IBM/ibm-block-csi-driver/blob/f77a552c283383cab8f7dc733326f60c94940223/controller/array_action/array_mediator_svc.py#L564)s

Reply:
done